### PR TITLE
Ci cd steps

### PR DESCRIPTION
**Pull request by wesyao**
- Add documentation for CI
- Add documentation for CD

### CODE REVIEW

File: ContinuousIntegrationAndDeployment.md [hunk 1]
```diff
@@ -40,4 +40,102 @@ On every successful build, continuous delivery is kicked off. Our reference arch
...
-`Stable` releases are typically a couple versions behind `canary` and the images are deemed stable for production. This will ensure that changes are highly available. Our solution requires an administrator to sign off on `stable` deployments. Rollbacks are easily accessible since each build is versioned.
\ No newline at end of file
+`Stable` releases are typically a couple versions behind `canary` and the images are deemed stable for production. This will ensure that changes are highly available. Our solution requires an administrator to sign off on `stable` deployments. Rollbacks are easily accessible since each build is versioned.
+
+## Creating Continuous Integration
+
+Each Repository requires 2 Build Definitions: one for PRs and one for master builds.
+
+**Prerequisites**
+- Kube Config must be on build/release machines.
```
Comment (aaron-schnieder) [inline ContinuousIntegrationAndDeployment.md]: How do you get kube config on the build/release machines in VSTS? Would be good to detail this one out.

File: ContinuousIntegrationAndDeployment.md [hunk 2]
```diff
@@ -40,4 +40,102 @@ On every successful build, continuous delivery is kicked off. Our reference arch
...
+- All dependencies must be installed on build/release machines.
```
Comment (aaron-schnieder) [inline ContinuousIntegrationAndDeployment.md]: Link to the dependencies.

File: ContinuousIntegrationAndDeployment.md [hunk 3]
```diff
@@ -40,4 +40,102 @@ On every successful build, continuous delivery is kicked off. Our reference arch
...
+    ![Add Build Tasks](/images/AddBuildTasks.png)
+
+6. Once the appropriate build has been setup, navigate to `Triggers` and connect the repo with this build as a pull request validator.
+    ![PR Validation](/images/PrValidation.png)
+
+### Master Builds
+
+1. Refer to steps 1-4 on section [Pull Request Builds](#Pull-Request-Builds)
+
+2. Add build variables by navigating to `Variables` and adding a `DockerImageName`, `MajorVersion`, and `MinorVersion`. The build will use these variables to semantically version the docker images.
```
Comment (aaron-schnieder) [inline ContinuousIntegrationAndDeployment.md]: "and add the following variables: "

Comment (aaron-schnieder) [inline ContinuousIntegrationAndDeployment.md]: Would be nice to explain how these variables are used a little deeper.

File: ContinuousIntegrationAndDeployment.md [hunk 4]
```diff
@@ -40,4 +40,102 @@ On every successful build, continuous delivery is kicked off. Our reference arch
...
+    - Name the artifact
+    - Select `Visual Studio Team Services/TFS` as the `Artifact publish location`.
+
+    ![Publish Build Artifact](/images/PublishArtifact.png)
+
+
+7. Navigate to `Triggers` and enable Continuous Integration. This build definition will now trigger when there are new changes in the master branch.
+    ![Enable Continuous Integration](/images/EnableContinuousIntegration.png)
+
+### Creating Continuous Deployment
```
Comment (aaron-schnieder) [inline ContinuousIntegrationAndDeployment.md]: I would make this an H2 for consistency.

**Comment by jobanputra** [comment]
General question: Is the more appropriate place of the CI/CD document in app and service repository?

**Comment by wesyao** [comment]
@aaron-schnieder Updating diagram in separate [PR](https://github.com/Microsoft/containerized-microservices-pipeline/pull/86)

@jobanputra I'm trying to keep the CI/CD general so users can add their own steps for each service

**Pull request merged**